src/G2/Equiv/Tactics.hs

```diff
@@ -931,7 +1050,8 @@ replaceMoreRestrictiveSubExpr' solver ns lemma@(Lemma { lemma_lhs = lhs_s, lemma
                 return rhs_e'
             Left _ -> do
                 let ns' = foldr HS.insert ns (bind e)
-                modifyChildrenM (replaceMoreRestrictiveSubExpr' solver ns' lemma s2) e
+                    sound' = lemmaSound ns' s2 lemma
+                modifyChildrenM (replaceMoreRestrictiveSubExpr' solver ns' lemma s2 (sound || sound')) e
```

Comment:
I don't follow what is happening with sound vs sound'?  Can you please add comment(s) explaining?